common/djangoapps/philu_commands/tests/helpers.py

```diff
@@ -0,0 +1,119 @@
+        """
+        super(TestHelpers, self).setUp()
+        self.course = CourseFactory.create(display_name='test course 1', run='Testing_course_1')
+        self.course.end = datetime.now(pytz.UTC) - timedelta(hours=2)
+
+    @patch('philu_commands.helpers.modulestore')
+    def test_generate_course_structure(self, mock_modulestore):
+        """
+        Test 'Successfully generate course structure'
+        """
+        mock_response = Mock(name='mock module store', **{'bulk_operations.return_value': MockModuleStoreTest(),
+                                                          'get_course.return_value': self.course})
+        mock_modulestore.return_value = mock_response
+        course_structure_data = generate_course_structure(self.course.id)
+        expected_course_object = {
+            'structure': {
+                'blocks': {
+                    u'i4x://org.0/course_0/course/Testing_course_1': {
+                        'block_type': 'course', 'graded': False, 'format': None,
+                        'usage_key': u'i4x://org.0/course_0/course/Testing_course_1',
+                        'children': [], 'display_name': u'test course 1'}},
+                'root': u'i4x://org.0/course_0/course/Testing_course_1'}, 'discussion_id_map': {}
+        }
+        assert course_structure_data == expected_course_object
+        ItemFactory.create(
+            parent=self.course,
+            parent_location=self.course.location,
+            category='discussion',
+            discussion_id='test_discussion',
+            discussion_category='Category discussion',
+            discussion_target='Target Discussion',
+        )
+        course_structure_data_2 = generate_course_structure(self.course.id)
+        expected_course_object_2 = {
+            'structure': {
+                'blocks': {
+                    u'i4x://org.0/course_0/course/Testing_course_1': {
+                        'block_type': 'course',
+                        'graded': False,
+                        'format': None,
+                        'usage_key': u'i4x://org.0/course_0/course/Testing_course_1',
+                        'children': [
+                            u'i4x://org.0/course_0/discussion/discussion_1'],
+                        'display_name': u'test course 1'
+                    },
+                    u'i4x://org.0/course_0/discussion/discussion_1': {
+                        'block_type': u'discussion',
+                        'graded': False,
+                        'format': None,
+                        'usage_key': u'i4x://org.0/course_0/discussion/discussion_1',
+                        'children': [],
+                        'display_name': u'discussion 1'
+                    }
+                },
+                'root': u'i4x://org.0/course_0/course/Testing_course_1'
+            },
+            'discussion_id_map': {
+                u'test_discussion': u'i4x://org.0/course_0/discussion/discussion_1'
```

Comment:
Instead of creating a new expected course object from scratch, resulting in code repetition, we can alter the existing course object that we created, like:

```
expected_course_object['structure']['blocks'][u'i4x://org.0/course_0/discussion/discussion_1'] = {
    'block_type': u'discussion',
    'graded': False,
    'format': None,
    'usage_key': u'i4x://org.0/course_0/discussion/discussion_1',
    'children': [],
    'display_name': u'discussion 1'
}
expected_course_object['discussion_id_map'] = {
    u'test_discussion': u'i4x://org.0/course_0/discussion/discussion_1'
}
```

Reply:
there is more change than this. and less repeatation. So i thought, it would be the best if i create a new object from the scratch. as it doesn't make a big difference.

But since we are trying to avoid any type of repeatation then I can also use `.update` method.